lib/utils/net_test.go

```diff
@@ -0,0 +1,70 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package utils
+
+import (
+	"testing"
+)
+
+func TestInferProxyPublicAddr(t *testing.T) {
+	tests := []struct {
+		name          string
+		fqdn          string
+		proxyDNSNames []string
+		expected      string
+	}{
+		{
+			name:          "Exact match",
+			fqdn:          "proxy.example.com",
+			proxyDNSNames: []string{"proxy.example.com"},
+			expected:      "proxy.example.com",
+		},
+		{
+			name:          "Tail match",
+			fqdn:          "app.proxy.example.com",
+			proxyDNSNames: []string{"proxy.example.com"},
+			expected:      "proxy.example.com",
+		},
+		{
+			name:          "No match returns first proxy DNS",
+			fqdn:          "nonexistent.domain.com",
+			proxyDNSNames: []string{"proxy.example.com"},
+			expected:      "proxy.example.com",
+		},
+		{
+			name:          "Empty FQDN returns empty string",
+			fqdn:          "",
+			proxyDNSNames: []string{"proxy.example.com"},
+			expected:      "",
+		},
+		{
+			name:          "Empty proxy list returns empty string",
+			fqdn:          "some.domain.com",
+			proxyDNSNames: []string{},
+			expected:      "",
+		},
```

Comment:
add test with multiple `proxyDNSNames` so it covers our logic here which infers one fqdn from slice https://github.com/gravitational/teleport/pull/52718/files#diff-8f50b520f8e5fe85fd56c4e8411ce2c91ecc43d03b417b08864ff8f979664c9aR66-R72

Reply:
Add test case for `host:port` proxy addr?